neurodiffeq/generators.py

```diff
@@ -961,3 +973,201 @@ def _internal_vars(self) -> dict:
             generator=self.generator,
         ))
         return d
+
+    def generator_loss(self):
+        if self._requires_training == True:
+          return self.generator.generator_loss
+        else:
+          return 0.0
+
+    def get_entropy(self):
+      return get_entropy_from_generator(self.generator)
+  
+
+
+class GaussianHistogram(nn.Module):
+    def __init__(self, bins, min, max, sigma):
+        super(GaussianHistogram, self).__init__()
+        self.bins = bins
+        self.min = min
+        self.max = max
+        self.sigma = sigma
+        self.delta = float(max - min) / float(bins)
+        self.centers = float(min) + self.delta * (torch.arange(bins).float() + 0.5)
+
+    def forward(self, x):
+        x = torch.unsqueeze(x, 0) - torch.unsqueeze(self.centers, 1)
+        x = torch.exp(-0.5*(x/self.sigma)**2) / (self.sigma * np.sqrt(np.pi*2)) * self.delta
+        x = x.sum(dim=1)
+        return x
+
+def get_entropy_from_generator(generator):
+  size = generator.size
+  t_min = generator.t_min
+  t_max = generator.t_max
```

Comment:
t_min and t_max are only available for `Generator1D`. So this function only works for 1D.